scripts/tests/test_suites.sh

```diff
@@ -38,22 +42,103 @@ usage() {
     echo "  -s CASE_NAME: runs single test case name (e.g. Test_TC_OO_2_2"
     echo "                for Test_TC_OO_2_2.yaml) (by default, all are run)"
     echo "  -w COMMAND: prefix all instantiations with a command (e.g. valgrind) (default: '')"
+    echo "  -n Use linux netns to isolate app and tool executables"
+    echo "  -c execute a netns cleanup and exit"
     echo ""
     exit 0
 }
 
-# read shell arguments
-while getopts a:d:i:hs:w: flag; do
+declare app_run_prefix=""
+declare tool_run_prefix=""
+
+netns_setup() {
+    if [[ `id -u` -ne 0 ]]; then
+        echo 'Running as non-root user, restarting in unshare environment'
```

Comment:
I slightly take issue with this warning. We **want** users to hit this path, this is the safe path. I would rather warn about other case

```
  if [[ `id -u` -ne 0 ]]; then
        echo 'Executing: ' $0 $INPUT_ARGS
        unshare --map-root-user -n -m $0 $INPUT_ARGS
        exit 0
    else
        # Don't run this as script as root outside of docker
        echo >&2 "Running as root, buckle your seatbelts..."
    fi
```

This check might also be better as

```
if [[ ! -w /run/netns ]]
```

and only mount that.

Reply:
Of course we would need to suppress the warning if we re-executed under unshare.

Reply:
Updated messaging a bit: kept executing, kept the `-r` logic and if that is not set, assume the 'root' uid is the real root and warn.